### PR TITLE
add the OpenGOAL project

### PR DESCRIPTION
**Pull request by romatthe**
Since other decompilation-related projects are now included in the list as well (`sm64ex`, `oot`, etc.), I believe OpenGOAL absolutely deserves a spot on this list!

### CODE REVIEW

**Comment by romatthe** [comment]
Since github/linguist doesn't define `GOAL`, I've removed it from the languages in which the project is written, though I'd still stay it's somewhat/semi correct.

We can leave it as is, or add a generic `Lisp` tag, I don't know which one is more appropriate.

File: games/o.yaml [hunk 1]
```diff
@@ -218,6 +218,32 @@
...
+  originals: 
+  - 'Jak and Daxter: The Precursor Legacy'
+  type: remake
+  repo: https://github.com/open-goal/jak-project
+  url: https://opengoal.dev/
+  development: active
+  status: playable
+  content: commercial
+  langs:
+  - C++
```
Comment (cxong) [inline games/o.yaml:231]:
```suggestion
  - C++
  - Common Lisp
```

**Review by cxong — APPROVED** [review]

**Comment by cxong** [comment]
if you want, you can raise an issue / PR to add GOAL to linguist, if it is worth it

**Comment by romatthe** [comment]
No, it's definitely not worth it to do that. It's fine as it is I believe, thanks for suggesting though.

**Pull request merged**